display/ssd1306.h

```diff
@@ -395,15 +404,19 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
 
       case SCREEN_ERROR_MESSAGE:
       case SCREEN_MESSAGE: {
-	uint32_t t;
+        uint32_t text_time;
         if (font_config.ProffieOSTextMessageDuration != -1) {
-	  t = font_config.ProffieOSTextMessageDuration;
-        } else if (font_config.ProffieOSFontImageDuration > 0) {
-	  t = font_config.ProffieOSFontImageDuration;
+          text_time = (font_config.ProffieOSTextMessageDuration == 0 ?
+                       SaberBase::sound_length * 1000 :
```

Comment:
We can't use SaberBase::sound_length here.
It's going to have basically random values outside of SB_*2.

Also, I'm not sure that we even want to do that SCREEN_ERROR_MESSAGE.

Reply:
oh i see. error message case falls through. hmm.
Would a new function based on ShowFileWithSoundLength() be appropriate for text like ShowTextWithSoundLength?
it's really only the preset "name" argument that could use it. 

Or not, and just set a hard time and don't sync to sound i guess. :(

Reply:
It seems reasonable.
Is this stuff related to the SB_On fix though?
If the answer is "not really", maybe split it into two PRs?

Reply:
Sure totally. The goal here was to rectify where the added code in SB_On() did not work if text duration wasn't set. It's really the comparison to `t_` that's not correct.